### DISCUSSION TITLE
How to open Secondary Window for custom ViewContribution

### DISCUSSION THREAD

**Discussion by aghayanlena** [eclipse-theia/theia#14269] [General]
Hi, 

Could you please guide us on how to open my custom view Contribution as a SecondaryWindow from Side Bar? 
Is there an example ? 

Thanks,
Lena

**Comment by msujew** [comment]
Hey @aghayanlena,

AFAIK you mostly just need to implement the [`ExtractableWidget` interface](https://github.com/eclipse-theia/theia/blob/master/packages/core/src/browser/widgets/extractable-widget.ts). Afterwards, the icon should appear on its own in the UI to open the widget in an individual window.
